File: 00938. Range Sum of BST/00938.go
// https://leetcode.com/problems/range-sum-of-bst

func rangeSumBST(root *TreeNode, low int, high int) (sum int) {
    if root == nil {
        return 0
    }

    if root.Val >= low && root.Val <= high {
        sum = root.Val
    }

    return sum + rangeSumBST(root.Left, low, high) + rangeSumBST(root.Right, low, high)
}

File: LeetCode/00938. Range Sum of BST/00938.c
// https://leetcode.com/problems/range-sum-of-bst

int rangeSumBST(struct TreeNode* root, int low, int high){
    if (root == NULL) {
        return 0;
    }

    int l = 0, r = 0, val = root->val;

    if (val < low) {
        r = rangeSumBST(root->right, low, high);
        val = 0;
    } else if (low <= val && val <= high) {
        l = rangeSumBST(root->left,  low, high);
        r = rangeSumBST(root->right, low, high);
    } else if (high < val) {
        l = rangeSumBST(root->left,  low, high);
        val = 0;
    }

    return l + r + val;
}
